src/Thesis/Ast.hs

```diff
@@ -33,3 +37,34 @@ data Select = Select {
   selectFrom :: Identifier,
   selectWhere :: Maybe Expr
 }
+
+selectToQuery :: Select -> (Query, [Parameter])
+selectToQuery (Select sAgg sFrom sWhere) = toQuery $ emit "SELECT " <> emitAggregation sAgg <> emitFrom sFrom <> emitWhere sWhere
+
+emitExpr :: Expr -> SqlPart
+emitExpr (Literal (Value v)) = emit $ pack $ show v
+emitExpr (Column identifier) = emitIdentifier identifier
+emitExpr (PrefixOp identifier expr) = emit identifier <> emitExpr expr
+emitExpr (PostfixOp identifier expr) = emitExpr expr <> emit identifier
+emitExpr (BinaryOp expr1 identifier expr2) = emitExpr expr1 <> emit identifier <> emitExpr expr2
+emitExpr (FunctionCall identifier exprs) = emit identifier <> emit "(" <> mconcat (emitExpr `map` exprs) <> emit ")"
+
+emitAggregation :: Aggregation -> SqlPart
+emitAggregation (Average expr) = emit "AVG(" <> emitExpr expr <> emit ") "
+emitAggregation (Sum expr) = emit "SUM(" <> emitExpr expr <> emit ") "
+emitAggregation (Count countExpr) = emit "COUNT(" <> case countExpr of
+  Star -> emit "*"
+  CountExpr expr -> emitExpr expr <> emit ") "
+  <> emit ") "
+
+emitWhere :: Maybe Expr -> SqlPart
+emitWhere Nothing = emit ""
```

Comment:
Since `SqlPart` is a monoid, you can use `mempty`, which is probably slightly cleaner.